core/node/da_clients/src/eigen/verifier.rs

```diff
@@ -0,0 +1,1347 @@
+use zksync_eth_client::{clients::PKSigningClient, EnrichedClientError, EnrichedClientResult};
+use zksync_types::{
+    web3::{self, BlockId, BlockNumber},
+    Address, U256, U64,
+};
+
+use super::blob_info::{BatchHeader, BlobHeader, BlobInfo, BlobQuorumParam, G1Commitment};
+
+#[async_trait::async_trait]
+pub trait VerifierClient: Sync + Send + std::fmt::Debug {
+    /// Returns the current block number.
+    async fn block_number(&self) -> EnrichedClientResult<U64>;
+
+    /// Invokes a function on a contract specified by `contract_address` / `contract_abi` using `eth_call`.
+    async fn call_contract_function(
+        &self,
+        request: web3::CallRequest,
+        block: Option<BlockId>,
+    ) -> EnrichedClientResult<web3::Bytes>;
+}
+
+#[async_trait::async_trait]
+impl VerifierClient for PKSigningClient {
+    async fn block_number(&self) -> EnrichedClientResult<U64> {
+        self.as_ref().block_number().await
+    }
+
+    async fn call_contract_function(
+        &self,
+        request: web3::CallRequest,
+        block: Option<BlockId>,
+    ) -> EnrichedClientResult<web3::Bytes> {
+        self.as_ref().call_contract_function(request, block).await
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum KzgError {
+    #[error("Kzg setup error: {0}")]
+    Setup(String),
+    #[error(transparent)]
+    Internal(#[from] rust_kzg_bn254::errors::KzgError),
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum ServiceManagerError {
+    #[error(transparent)]
+    EnrichedClient(#[from] EnrichedClientError),
+    #[error("Decoding error: {0}")]
+    Decoding(String),
+    #[cfg(test)]
+    #[error("Parsing error: {0}")]
+    Parsing(String),
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum VerificationError {
+    #[error(transparent)]
+    ServiceManager(#[from] ServiceManagerError),
+    #[error(transparent)]
+    Kzg(#[from] KzgError),
+    #[error("Wrong proof")]
+    WrongProof,
+    #[error("Different commitments: expected {expected:?}, got {actual:?}")]
+    DifferentCommitments {
+        expected: G1Affine,
+        actual: G1Affine,
+    },
+    #[error("Different roots: expected {expected:?}, got {actual:?}")]
+    DifferentRoots { expected: String, actual: String },
+    #[error("Empty hash")]
+    EmptyHash,
+    #[error("Different hashes: expected {expected:?}, got {actual:?}")]
+    DifferentHashes { expected: String, actual: String },
+    #[error("Wrong quorum params: {blob_quorum_params:?}")]
+    WrongQuorumParams { blob_quorum_params: BlobQuorumParam },
+    #[error("Quorum not confirmed")]
+    QuorumNotConfirmed,
+    #[error("Commitment not on curve: {0}")]
+    CommitmentNotOnCurve(G1Affine),
+    #[error("Commitment not on correct subgroup: {0}")]
+    CommitmentNotOnCorrectSubgroup(G1Affine),
+    #[error("Link Error: {0}")]
+    LinkError(String),
+}
+
+/// Configuration for the verifier used for authenticated dispersals
+#[derive(Debug, Clone)]
+pub struct VerifierConfig {
```

Comment:
This struct doesn't make sense to me.

- It is public, but it shouldn't be because it's only used in a `pub(crate)` constructor. (I believe we had this discussion already.)
- Some of its fields (like `private_key` or `chain_id`) are only used locally in `RawEigenClient::new()`, so can be easily eliminated. In the particular case of `private_key`, this use may get removed anyway (see another comment about signing clients).
- The config is stored as a field in `Verifier`, but only 2 fields from it are read (`svc_manager_addr` and `settlement_layer_confirmation_depth`).

I'd suggest to remove this config and instead store only data necessary for `Verifier` function. As for args, could make sense to pass `&EigenConfig`.